bot/util.go

```diff
@@ -265,26 +275,60 @@ func NodeID() string {
 
 // RefreshStatus updates the provided sessions status according to the current status set
 func RefreshStatus(session *discordgo.Session) {
-	var streamingURL string
-	var status string
-	err1 := common.RedisPool.Do(radix.Cmd(&streamingURL, "GET", "status_streaming"))
-	err2 := common.RedisPool.Do(radix.Cmd(&status, "GET", "status_name"))
+	var streamingUrl string
+	var statusText string
+	var gameType string
+	var statusTypeStr string
+	var statusType discordgo.Status
+	err1 := common.RedisPool.Do(radix.Cmd(&streamingUrl, "GET", "status_streaming_url"))
+	err2 := common.RedisPool.Do(radix.Cmd(&statusText, "GET", "status_name"))
+	err3 := common.RedisPool.Do(radix.Cmd(&gameType, "GET", "status_game_type"))
+	err4 := common.RedisPool.Do(radix.Cmd(&statusTypeStr, "GET", "status_type"))
+
 	if err1 != nil {
-		logger.WithError(err1).Error("failed retrieiving bot streaming status")
+		logger.WithError(err1).Error("failed retrieving bot streaming status")
 	}
 	if err2 != nil {
-		logger.WithError(err2).Error("failed retrieiving bot status")
+		logger.WithError(err2).Error("failed retrieving bot status")
+	}
+	if err3 != nil {
+		logger.WithError(err3).Error("failed retrieving bot game type")
+	}
+	if err4 != nil {
+		logger.WithError(err4).Error("failed retrieving bot status type")
 	}
 
-	if streamingURL != "" {
-		session.UpdateStreamingStatus(0, status, streamingURL)
+	switch statusTypeStr {
+	case "online":
+		statusType = discordgo.StatusOnline
+	case "idle":
+		statusType = discordgo.StatusIdle
+	case "dnd":
+		statusType = discordgo.StatusDoNotDisturb
+	case "offline":
+		statusType = discordgo.StatusOffline
+	default:
+		statusType = discordgo.StatusOnline
+	}
+	if streamingUrl != "" || gameType == "streaming" {
+		session.UpdateStreamingStatus(statusText, statusType, streamingUrl)
 	} else {
-		session.UpdateStatus(0, status)
+		switch gameType {
+		case "playing":
+			session.UpdatePlayingStatus(statusText, statusType)
+		case "listening":
+			session.UpdateListeningStatus(statusText, statusType)
+		case "watching":
+			session.UpdateWatchingStatus(statusText, statusType)
+		case "competing":
+			session.UpdateCompetingStatus(statusText, statusType)
+		default: // covers "custom"
+			session.UpdateStatus(statusText, statusType)
+		}
```

Comment:
Wouldn't it be easier to change the function signature of
`session.UpdateStatus` to also accept the `gameType`? That way we can
avoid this massive switch-case statement that doesn't do much else.

Though, we'd probably still have to have the variants we're calling here
elsewhere, probably unexported. In my opinion the user of this library
(even if it's just us) should not have to worry about calling the proper
function, we can handle that for them.

Reply:
Are you saying move the switch-case to `UpdateStatus`? Could do that, was just following the format that was already there (which, admittedly, was only streaming or not streaming)

Reply:
> Are you saying move the switch-case to `UpdateStatus`? Could do that,
> was just following the format that was already there (which,
> admittedly, was only streaming or not streaming)

Yeah, I think that'd be easier to follow.